cdap-ui/app/cdap/components/PipelineNodeGraphs/PipelineNodeMetricsGraph.js

```diff
@@ -202,12 +322,108 @@ export default class PipelineNodeMetricsGraph extends Component {
+  */
+  renderProcesstimeTable = () => {
+    let totalProcessingTime, validRecords;
+    Object
+      .keys(this.state.processTimeMetrics)
+      .forEach(metric => {
+        if (metric.match(/user.*.process.time.total/)) {
+          totalProcessingTime = this.state.processTimeMetrics[metric] / 1000000;
+        }
+        if (this.isPluginSource() && metric.match(/user.*.records.in/)) {
+          validRecords = this.state.processTimeMetrics[metric];
+          return;
+        }
+        if (this.isPluginSink() && metric.match(/user.*.records.out/)) {
+          validRecords = this.state.processTimeMetrics[metric];
+          return;
+        }
+        if (metric.match(/user.*.records.out/)) {
+          validRecords = this.state.processTimeMetrics[metric];
+        }
+      });
+    let processRate, avgProcessingTime;
+    if (isNil(validRecords) || isNil(totalProcessingTime)) {
+      processRate = T.translate('commons.notAvailable');
+      avgProcessingTime = T.translate('commons.notAvailable');
+    } else {
+      processRate = parseFloat(validRecords / totalProcessingTime, 10).toFixed(3);
+      avgProcessingTime = ((1 / processRate) * 1000000); // Because total processing time is in seconds
+    }
+    const renderMicroSeconds = (time) => {
+      if (time < 1000) {
+        return `${parseFloat(time / 1000, 10).toFixed(2)} ${T.translate('commons.milliSecondsShortLabel')}`;
+      }
+      return humanReadableDuration(
+        Math.ceil(this.state.processTimeMetrics[REGEXTOLABELLIST[0].id] / 1000000)
+      ) || T.translate('commons.notAvailable');
+    };
+    return (
+      <div className="process-time-table-container">
+        <table className="table table-sm">
+          <thead>
+            <tr>
+              <th>{T.translate(`${PREFIX}.processTimeTable.recordOutPerSec`)}</th>
+              <th>{T.translate(`${PREFIX}.processTimeTable.minProcessTime`)}</th>
+              <th>{T.translate(`${PREFIX}.processTimeTable.maxProcessTime`)}</th>
+              <th>{T.translate(`${PREFIX}.processTimeTable.stddevProcessTime`)}</th>
+              <th>{T.translate(`${PREFIX}.processTimeTable.avgProcessTime`)}</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td>{processRate}</td>
+              <td>{renderMicroSeconds(this.state.processTimeMetrics[REGEXTOLABELLIST[0].id])}</td>
+              <td>{renderMicroSeconds(this.state.processTimeMetrics[REGEXTOLABELLIST[1].id])}</td>
+              <td>{renderMicroSeconds(this.state.processTimeMetrics[REGEXTOLABELLIST[2].id])}</td>
+              <td>{renderMicroSeconds(avgProcessingTime)}</td>
+            </tr>
+          </tbody>
+        </table>
+      </div>
+    );
+  };
+
+  isPluginSource = () => {
+    return ['batchsource', 'realtimesource', 'streamingsource'].indexOf(this.props.plugin.type) !== -1;
+  };
+
+  isPluginSink = () => {
+    return ['batchsink', 'realtimesink', 'sparksink'].indexOf(this.props.plugin.type) !== -1;
+  };
```

Comment:
Maybe extract these 2 functions out to `helpers.js`? I think this isn't or won't be the first time you'll need to use these checks.